src/main/docs/serializers.adoc

```diff
@@ -0,0 +1,265 @@
+----
+
+In short this is what the above serializer does: Whenever it gets a string(the fruit name, in our application) to be
+serialized it checks if there is a mapping that exists already for that name in *stringIdMap*.
+If yes, the mapped integer is retrieved from the map and that integer value is serialized.
+If a mapping is not found, we generate a new *id* for the new fruit name add it to both the maps that we preserve
+(*stringIdMap* and *idStringMap*) and then serialize this newly generated id.
+Now on deserialization, the same *idStringMap* map is used to retrieve the fruit names from the deserialised integer
+values.
+
+So in the above serializer, the *idStringMap*, *stringIdMap* and the *id* constitutes the _state_ of the serializer.
+The serialization and deserialization depends on this state and would not work properly without that state.
+This serializer has the single-argument constructor making it fit to be used with transient caches.
+So now that we have a state-full serializer understanding the idea of _transient_ and _persistent_ serializers would be
+simpler.
+
+Here is a sample code that uses the `DumbTransientStringSerializer` with a cache:
+
+[source,java,indent=0]
+----
+include::{sourcedir}/test/java/org/ehcache/demo/serializer/SerializersDemo.java[tag=transientSerializerGoodSample]
+----
+
+<1> Create a multi-tiered cache that depends on key and value serialization.
+<2> Configure a serializer for the values. The `DumbTransientStringSerializer` in this case. For the sake of simplicity
+    we have omitted key serializer. Since one is not provided explicitly, ehcache would provide default serializers to
+    perform the key serialization.
+<3> Verify that the cache/serializer works.
+
+In the previous section we demonstrated the use of a transient serializer.
+We used that serializer with a transient cache and everything works just fine.
+Now imagine what would happen if we use the same serializer with a persistent cache.
+Everything would work as long as your application is running. Once you close the cache manager or end the application 
+the data associated with the cache will be persisted so that the same data will be available on a restart.
+But there is a serious problem. The following piece of code would demonstrate that:
+
+[source,java,indent=0]
+----
+include::{sourcedir}/test/java/org/ehcache/demo/serializer/SerializersDemo.java[tag=transientSerializerBadSample]
+----
+
+<1> Create a cache configuration with persistent disk tier.
+<2> Configure the `LocalPersistenceService` for the cache manager.
+<3> Retrieve the cache.
+<4> Populate data.
+<5> Verify that everything works.
+<6> Close the cache manager.
+<7> Reinitialize the cache manager.
+<8> Retrieve the cache.
+<9> Retrieve a cached/persisted value.
+
+The above piece of code would fail in the cache creation step since the serializer provided does not meet the 2-arg 
```

Comment:
All of this as a single paragraph is just too much. Split it in different paragraphs, telling your story.